executor/linux/build.go

```diff
@@ -456,13 +453,13 @@ func (c *client) ExecBuild(ctx context.Context) error {
 	r := c.repo
 	e := c.err
 
-	b.SetStatus(constants.StatusSuccess)
-	c.build = b
-
 	defer func() {
 		// NOTE: When an error occurs during a build that does not have to do
 		// with a pipeline we should set build status to "error" not "failed"
 		// because it is worker related and not build.
+		b.SetStatus(constants.StatusSuccess)
+		c.build = b
```

Comment:
Should we move the comment below these two lines?

```suggestion
		b.SetStatus(constants.StatusSuccess)
		c.build = b
		
		// NOTE: When an error occurs during a build that does not have to do
		// with a pipeline we should set build status to "error" not "failed"
		// because it is worker related and not build.
```

Reply:
Sure! I wasn't exactly sure where to place it